### DISCUSSION TITLE
Can't setAuthCookies on HTTPS

### DISCUSSION THREAD

**Discussion by einaralex** [gladly-team/next-firebase-auth#229] [Q&A]
Hey, I've been struggling with this for a while.

Everything works perfectly on localhost with http.

On Production with HTTPS  though, when logging in the setAuthCookies returns and error: `The request is missing an Authorization header value`

I've tried toggling the `signed` and `secure` flags true/false.

Here is my setup: https://github.com/kontaktar/www/blob/firebase/src/lib/initAuth.ts

I don't quite get at what point `headers.authorization` is set

**Comment by einaralex** [comment]
Hey @kmjennison !

At what point is the Authorization header set in this package?

**Reply by kmjennison** [reply]
@einaralex The Authorization header is included by default when calling the login API endpoint on Firebase auth change:
https://github.com/gladly-team/next-firebase-auth/blob/main/src/useFirebaseUser.js#L17

I suggest you first double-check the Authorization header value is included in your request from the client. Then, you might also want to confirm your server is passing forward the Authorization header value, which is currently [required for `setAuthCookies`](https://github.com/gladly-team/next-firebase-auth/blob/3174d69dbaa6fc1035c7aeeae0a2aac3c014e565/src/setAuthCookies.js#L10).

**Reply by kmjennison** [reply]
Related: #233
